Issue 57 - Use indicatif instead of spinners library for command line waiting spinner

byblakeorriver opened the pull request:
Signed-off-by: bwilloughby <bwilloughby@demark.com>

byblakeorriver commented:
@brooksmtownsend I think this one is ready to go. Let me know if you have any concerns with the way I did it.

brooksmtownsend reviewed (changes requested):
Overall this looks awesome! Cleans up the code and our dependencies 😄 

I did notice one quirk about using the `reg` functions, it looks like there is an extra line with one dot that gets printed, see picture.
<img width="471" alt="image" src="https://user-images.githubusercontent.com/12040685/157055498-b36956db-2b21-47ee-b990-ea296ff04d9a.png">

I don't see the same thing with the `ctl` commands, perhaps it's due the the info log that I reference in my review?

I also tested this on Windows in Powershell and it worked well, just see my comment around the blue spinner.

byblakeorriver commented on src/reg.rs:
```diff
@@ -134,14 +134,12 @@ pub(crate) async fn handle_pull(
+    let spinner = Spinner::new(&output_kind);
+    let msg = format!(" Downloading {} ...", image.whole());
+    spinner.update_spinner_message(msg.clone());
+    info!("{}", msg);
```
I thought these might cause some trouble. I didn't know if the logging was there for any particular reason so I kept them. I will get it cleaned up.

byblakeorriver commented:
@brooksmtownsend can you give this last commit a try when you get a chance? I tried to recreate your issue, but was not able to get it to occur. I tried on two terminals (alacritty and the default pop_os terminal). I also tried both of those using the fish shell to see if that could maybe reproduce it. 

I did however find an issue if the terminal window is really small. It would print the message over and over again. I fixed that using the wide_msg template setting in the indicatif lib. Maybe that could have something to do with it.

I did get the color changed. I couldn't find a specific option for the gray, but I did find a way to make it bold and dim. Let me know what you think. I also added a prefix of (>>>) not sure if that is ok with you. It could be anything. An emoji or just a couple spaces.

byblakeorriver commented:
I was able to find a way to choose any of these colors: https://www.ditig.com/256-colors-cheat-sheet

If you don't like the bold dim version, I change the colors, just need some artistic advice on which one to choose.

brooksmtownsend reviewed (approved):
This is great, and I didn't even know we could fix the printing over and over again issue 😄
